nni/compression/pytorch/speedup/jit_translate.py

```diff
@@ -422,6 +424,266 @@ def parse_aten_schema(schema: str):
 
     return positional_num, keyword_list, special_treat
 
+@lru_cache
+def parse_aten_schema_version_1_8_x(schema: str):
+    """
+    Parse the schema, to positional_num and keyword_list, and detect if the argument should be specially treated.
+    Cannot use 'torch._C.parse_schema' because 'torch._C.Argument' has no 'kwarg_only' on pytorch v1.8.x
+    Using a lexer-parser like method to parse it.
+    Re-write from torch/csrc/jit/frontend/function_schema_parser.cpp
```

Comment:
if this function a mirror rewrite from this `cpp` file?

Reply:
It's not totally equal to the original function in cpp. Actually, the original parser in that cpp file is too specific and too long (5k+ lines). I wrote a much simpler syntax analyzer, deleted some error check and specific type check to shorten the code.